cmd/docker-app/deploy.go

```diff
@@ -77,7 +140,19 @@ func runDeploy(dockerCli command.Cli, flags *pflag.FlagSet, appname string, opts
 	if stackName == "" {
 		stackName = internal.AppNameFromDir(app.Name)
 	}
-	if app.Source.ShouldRunInsideDirectory() {
+	if len(app.Attachments()) > 0 && internal.Experimental == "on" {
+		tempDir, err := ioutil.TempDir("", "app-deploy")
+		if err != nil {
+			return errors.Wrap(err, "failed to create temporary directory")
+		}
+		defer os.RemoveAll(tempDir)
+		if err := renderAttachments(app, tempDir, d, opts.deployNoRenderAttachment); err != nil {
+			return err
+		}
+		if err := os.Chdir(tempDir); err != nil {
```

Comment:
If I read correctly the code, the temp dir only contains the rendered attachments. The `Chdir` to the temporary directory is made just before the `stack.RunDeploy` invocation, so I guess it modifies the deploy behavior?